### PR TITLE
Support CIS2 backchannel logout

### PR DESCRIPTION
**Pull request by misaka**
This is required by CIS2 technical conformance. It adds an callback `/auth/cis2_logout` which CIS2 will use to logout a user session.

Testing this in a real environment is tricky, it's not even clear yet how to get this endpoint registered with CIS2.

### CODE REVIEW

File: app/lib/nhs/cis2.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            Rails.cache.delete("cis2:jwks")
+          end
+          Rails
+            .cache
+            .fetch("cis2:jwks") do
+              @cache_last_update = Time.zone.now.to_i
+              jwks_hash = JSON.parse(Faraday.get(jwks_uri).body)
+              jwks = JWT::JWK::Set.new(jwks_hash)
+              jwks.select! { |key| key[:use] == "sig" }
+              jwks
```
Comment (thomasleese) [inline app/lib/nhs/cis2.rb:22]:
```suggestion
              jwks.select { |key| key[:use] == "sig" }
```

**Review by thomasleese — APPROVED** [review]
Just one small comment, but otherwise looks good.

File: app/lib/nhs/cis2.rb (hunk 1 above)
Comment (misaka) [reply app/lib/nhs/cis2.rb:22]: Ya, agreed, that makes more sense.

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -2,9 +2,11 @@
 
 class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
   include AuthenticationConcern
+  include CIS2LogoutConcern
 
   skip_before_action :authenticate_user!
   skip_after_action :verify_policy_scoped
+  skip_before_action :verify_authenticity_token, only: [:cis2_logout]
```
[the remark this answers is not in the document]
Comment (misaka) [reply app/controllers/users/omniauth_callbacks_controller.rb:9]: Dismissed because we validate this request by verifying the logout JWT provided.

**Review by tvararu — APPROVED** [review]

**Pull request merged**